src/cron.rs

```diff
@@ -131,54 +131,61 @@ pub fn update_nodes(
     }
 
     // Compute which nodes changed their state, also update node names in DB
+    let online_nodes = cur_nodes_map.values().filter(|data| data.online).count();
     let changed : Vec<(String, NodeData)> = db.transaction::<_, anyhow::Error, _>(|| {
         let mut changed = Vec::new();
 
-        // Go over every node in the database
-        let db_nodes = nodes::table.load::<models::NodeQuery>(db)?;
-        for db_node in db_nodes.into_iter() {
-            let (id, db_data) = model_to_node_data(db_node);
-            if let Some(cur_data) = cur_nodes_map.remove(&id) {
-                // We already know this node.
-                // Did it change?
-                if cur_data != db_data {
-                    // Update in database
-                    diesel::update(nodes::table.find(id.as_str()))
-                        .set((
-                            nodes::name.eq(cur_data.name.as_str()),
-                            nodes::online.eq(cur_data.online)
-                        ))
+        if online_nodes < config.ui.min_online_nodes_on_map {
+            // don't do anything if nearly all nodes are offline
+            // The cronjob will print this and the cronjob will send a notification email to the admin:
+            println!("Only {} nodes are online - the map seems to be showing all nodes offline", online_nodes);
```

Comment:
No, you can't just print to stdout, that makes no sense. This would appear on the server. You want the text to be sent as response to the client. For this you'll need to change the return type of `update_nodes`, to return a `bool` saying whether everything is okay, and then in

https://github.com/freifunk-saar/ff-node-monitor/blob/44306142ec0aa190277cd4d71c8b8307f11f2392/src/routes.rs#L169

you translate that into returning an appropriate response to the HTTP request.

Reply:
We will do that later in another step, first lets get this running

Reply:
Not sure what you mean by "later step", but I don't think it makes sense to land the PR with a `println!`. Of course feel free to test things more etc before getting back to this, but please leave this conversation marked as unresolved, as this should be taken care of before merging.

Reply:
I added a comment in the code, I think we could merge this without a warning implemented here (It is way above my Rust skills, right now ;-) )  by "later step" I meant a new PR later.

Reply:
For a tool whose primary purpose is sending warnings about offline nodes, I don't think we should stop sending those warnings without any notification. So I think this has to happen in this PR.

But maybe once everything else is done I can find some time to do that part.